merino-admin/suggestion_providers/models.py

```diff
@@ -0,0 +1,330 @@
+        help_text=INNER_PROVIDER_HELP_TEXT,
+    )
+
+    def get_type(self):
+        return "memory_cache"
+
+
+class RedisCacheConfig(ProviderConfig):
+    help_text = """
+        Cache suggestions in a cluster-wide Redis store.
+    """
+
+    default_ttl: models.DurationField = models.DurationField(
+        blank=True,
+        null=True,
+        help_text=f"""
+            The default TTL to assign to a cache entry if the underlying
+            provider does not provide one. {DURATION_FORMAT_HELP_TEXT}
+            {VALUE_WITH_DEFAULT_HELP_TEXT}
+        """,
+    )
+    default_lock_timeout: models.DurationField = models.DurationField(
+        blank=True,
+        null=True,
+        help_text=f"""
+            The default TTL for in-memory locks to prevent multiple update
+            requests from being fired at providers at the same time.
+            {DURATION_FORMAT_HELP_TEXT} {VALUE_WITH_DEFAULT_HELP_TEXT}
+        """,
+    )
+    inner: models.ForeignKey = models.ForeignKey(
+        ProviderConfig,
+        on_delete=models.SET_NULL,
+        null=True,
+        related_name="redis_cache",
+        help_text=INNER_PROVIDER_HELP_TEXT,
+    )
+
+    def get_type(self):
+        return "redis_cache"
+
+
+class MultiplexerConfig(ProviderConfig):
+    help_text = """
+        Combine suggestions from multiple providers.
+    """
+
+    providers: models.ManyToManyField = models.ManyToManyField(
+        ProviderConfig,
+        related_name="multiplexer",
+        help_text="""
+            The providers to generate suggestions with. All providers will be
+            run in parallel for all requests.
+        """,
+    )
+
+    def get_type(self):
+        return "multiplexer"
+
+
+class TimeoutConfig(ProviderConfig):
+    help_text = """
+        Wait a configurable amount of time for suggestions, and if the inner
+        provider takes too long, return an empty result.
+    """
+
+    max_time: models.DurationField = models.DurationField(
+        help_text=f"""
+        After this much time, an empty response will be returned and the request
+        to the inner provider will be cancelled. {DURATION_FORMAT_HELP_TEXT}
+    """
+    )
+    inner: models.ForeignKey = models.ForeignKey(
+        ProviderConfig,
+        on_delete=models.SET_NULL,
+        null=True,
+        related_name="timeout",
+        help_text=INNER_PROVIDER_HELP_TEXT,
+    )
+
+    def get_type(self):
+        return "timeout"
+
+
+class FixedConfig(ProviderConfig):
+    help_text = """
+        A debug provider that always return a suggestion with dummy values and
+        configurable fixed string as a title.
+    """
+
+    value: models.CharField = models.CharField(
+        max_length=256,
+        help_text="The value to use for the fixed title of the suggestion.",
+    )
+
+    def get_type(self):
+        return "fixed"
+
+
+class Keyword(models.Model):
+    help_text = """
+        A keyword that can be used to match suggestions for filtering.
```

Comment:
Nit: the keyword filters will be matched against the title of the suggestion, not the suggestion per se.